src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -63,6 +61,13 @@ static String agreementEntriesTable(TenantPgPool pool) {
     return pool.getSchema() + ".agreement_entries";
   }
 
+  static void failHandler(RoutingContext ctx) {
+    Throwable t = ctx.failure();
+    // both semantic errors and syntax errors are from same pile.. Choosing 400 over 422.
+    int statusCode = t.getClass().getName().startsWith("io.vertx.ext.web.validation") ? 400 : 500;
+    failHandler(statusCode, ctx, t.getMessage());
```

Comment:
I always love to see this — there's nothing more frustrating for a front-end developer than a response that _just_ says "400 Bad Request"!